sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java

```diff
@@ -148,6 +157,48 @@ public static void recordOperation(
         );
     }
 
+    private static boolean hasAnyActualMeterRegistry() {
+
+        Instant nowSnapshot = Instant.now();
+        DescendantValidationResult snapshot = lastDescendantValidation;
+        if (nowSnapshot.isBefore(snapshot.getExpiration())) {
+            return snapshot.getResult();
+        }
+
+        synchronized (lockObject) {
+            snapshot = lastDescendantValidation;
+            if (nowSnapshot.isBefore(snapshot.getExpiration())) {
+                return snapshot.getResult();
+            }
+
+            DescendantValidationResult newResult = new DescendantValidationResult(
+                nowSnapshot.plus(10, ChronoUnit.SECONDS),
+                hasAnyActualMeterRegistryCore(compositeRegistry, 1)
```

Comment:
why the need for this recursive function?
Mainly concerned about the memory stack that this will create and specially when it is being called inside synchronized block.

Reply:
See - above - because a CMR could contain another CMR - but I have limited the recursion (will bail out when depth is > 100 - we could further reduce it - I think even 5 or 10 should be ok).

Reply:
Depth restriction is in line 186